.github/workflows/codespell.yml

```diff
@@ -0,0 +1,25 @@
+# Codespell configuration is within .codespellrc
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
```

Comment:
This one is fine: this is GitHub's responsibility and if this does something malicious, the whole Internet will break anyway. So please keep this as `@v4` so it's easier to eyeball that we're doing something canonical.